src/app/layout/qgslayout3dmapwidget.cpp

```diff
@@ -0,0 +1,93 @@
+/***************************************************************************
+  qgslayout3dmapwidget.cpp
+  --------------------------------------
+  Date                 : August 2018
+  Copyright            : (C) 2018 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgslayout3dmapwidget.h"
+
+#include "qgisapp.h"
+#include "qgs3dmapcanvas.h"
+#include "qgs3dmapcanvasdockwidget.h"
+#include "qgs3dmapsettings.h"
+#include <QMenu>
+
+QgsLayout3DMapWidget::QgsLayout3DMapWidget( QgsLayoutItem3DMap *map3D )
+  : QgsLayoutItemBaseWidget( nullptr, map3D )
+  , mMap3D( map3D )
+{
+  setupUi( this );
+
+  //add widget for general composer item properties
+  mItemPropertiesWidget = new QgsLayoutItemPropertiesWidget( this, map3D );
+  mainLayout->addWidget( mItemPropertiesWidget );
+
+  mMenu3DCanvases = new QMenu( this );
+  mCopySettingsButton->setMenu( mMenu3DCanvases );
+  connect( mMenu3DCanvases, &QMenu::aboutToShow, this, [ = ]
+  {
+    const QList<Qgs3DMapCanvasDockWidget *> lst = QgisApp::instance()->findChildren<Qgs3DMapCanvasDockWidget *>();
+    mMenu3DCanvases->clear();
+    for ( auto dock : lst )
+    {
+      QAction *a = mMenu3DCanvases->addAction( dock->mapCanvas3D()->objectName(), this, &QgsLayout3DMapWidget::copy3DMapSettings );
+      // need to use a custom property for identification because Qt likes to add "&" to the action text
+      a->setProperty( "name", dock->mapCanvas3D()->objectName() );
+    }
+    if ( lst.isEmpty() )
+    {
+      mMenu3DCanvases->addAction( tr( "No 3D maps defined" ) )->setEnabled( false );
+    }
+  } );
+
+  QgsCameraPose pose = mMap3D->cameraPose();
+  mCenterXSpinBox->setValue( pose.centerPoint().x() );
+  mCenterYSpinBox->setValue( pose.centerPoint().y() );
+  mCenterZSpinBox->setValue( pose.centerPoint().z() );
+  mDistanceToCenterSpinBox->setValue( pose.distanceFromCenterPoint() );
+  mPitchAngleSpinBox->setValue( pose.pitchAngle() );
+  mHeadingAngleSpinBox->setValue( pose.headingAngle() );
+
+  QList<QgsDoubleSpinBox *> lst;
+  lst << mCenterXSpinBox << mCenterXSpinBox << mCenterXSpinBox << mDistanceToCenterSpinBox << mPitchAngleSpinBox << mHeadingAngleSpinBox;
```

Comment:
Pasting issue, the 2nd and 3rd mCenterXSpinBox should be mCenterYSpinBox and mCenterZSpinBox :)